### PR TITLE
Adding hive-conf support to Flotilla for EMR/EKS

### PR DESCRIPTION
**Pull request by ujjwalsfix**
https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-spark-configure.html

### CODE REVIEW

File: execution/engine/emr_engine.go [hunk 1]
```diff
@@ -107,22 +107,32 @@ func (emr *EMRExecutionEngine) Execute(executable state.Executable, run state.Ru
...
 		"spark.eventLog.enabled":                    aws.String(fmt.Sprintf("true")),
 	}
+	hiveDefaults := map[string]*string{}
 
 	for _, k := range run.SparkExtension.ApplicationConf {
-		properties[*k.Name] = k.Value
+		sparkDefaults[*k.Name] = k.Value
 	}
+
+	for _, k := range run.SparkExtension.HiveConf {
```
Comment (nssalian) [inline execution/engine/emr_engine.go:124]: Is this pulling AWS's default HiveConf? And our overrides in the operator should be able to replace these?

File: execution/engine/emr_engine.go [hunk 2]
```diff
@@ -107,22 +107,32 @@ func (emr *EMRExecutionEngine) Execute(executable state.Executable, run state.Ru
...
-	properties := map[string]*string{
+	sparkDefaults := map[string]*string{
 		"spark.kubernetes.driver.podTemplateFile":   emr.driverPodTemplate(executable, run, manager),
 		"spark.kubernetes.executor.podTemplateFile": emr.executorPodTemplate(executable, run, manager),
 		"spark.kubernetes.container.image":          &run.Image,
 		"spark.eventLog.dir":                        aws.String(fmt.Sprintf("s3a://%s/%s", emr.s3LogsBucket, emr.s3EventLogPath)),
 		"spark.history.fs.logDirectory":             aws.String(fmt.Sprintf("s3a://%s/%s", emr.s3LogsBucket, emr.s3EventLogPath)),
 		"spark.eventLog.enabled":                    aws.String(fmt.Sprintf("true")),
 	}
+	hiveDefaults := map[string]*string{}
```
Comment (nssalian) [inline execution/engine/emr_engine.go:118]: Why is this empty? I think we should add at least the metastore uri here.

File: execution/engine/emr_engine.go (hunk 1 above)
Comment (ujjwalsfix) [reply execution/engine/emr_engine.go:124]: no defaults here.

**Review by derekpbennett — APPROVED** [review]
Seems like the right structure to pass this to EMR

**Pull request merged**